keras/applications/imagenet_utils.py

```diff
@@ -58,6 +49,75 @@ def preprocess_input(x, data_format=None, mode='caffe'):
     return x
 
 
+def _bias_add_match_type(x, bias):
```

Comment:
Because a single call to `bias_add` can handle all cases, you won't need a separate function to the casting. You can do it in the main function.